### PR TITLE
add calibration_setup_helper program

### PR DESCRIPTION
**Pull request by k-okada**
- setup helper program for generating calibration package for a robot
- here is how I checked the program
  - run baxter simulaiton

```
<launch>
  <include file="$(find baxter_gazebo)/launch/baxter_world.launch" />
  <anode pkg="baxter_interface" type="joint_trajectory_action_server.py"
        args="--limb both" name="joint_trajectory_action_server" />
  <node pkg="image_proc" type="image_proc" name="camera_rgb_image_proc"
        ns="/camera/rgb" />
  <node pkg="topic_tools" type="relay" args="/robot/joint_states /joint_states"
        name="joint_states_relay" />
  <node pkg="image_view" type="image_view" name="image_view_rect" >
    <remap from="image" to="/camera/rgb/image_rect" />
  </node>

</launch>
```
- run setup helper under <catkin_ws>/src/ directory

```
(cd src; rosrun calibration_setup_helper calibration_setup_helper.py
        `rospack find baxter_description`/urdf/baxter.urdf
        --base-link 'torso' --arm-end-link 'right_gripper_base'
        --head-end-link 'camera_rgb_frame'
        --arm-controller=/robot/limb/right/command
        --head-controller=/robot/limb/head/command)
```
- run catkin_make to register generated baxter_calibration package
- run make_sample.py program to correct calib posture

```
rosrun baxter_calibration make_samples.py
```

   use C-c to step
- run capture data program

```
roslaunch baxter_calibration capture_data.launch
```
- run calibration program

```
rosrun baxter_calibration calibrate_baxter.sh
```

### CODE REVIEW

**Comment by k-okada** [comment]
you need to include foolowing PRs to run examples
- https://github.com/RethinkRobotics/baxter_interface/pull/37
- https://github.com/ros/urdfdom/pull/46
- https://github.com/ros-perception/calibration/pull/27

**Comment by k-okada** [comment]
It may not work well, please do not merge yet

**Comment by k-okada** [comment]
I think it is now ok to merge

**Comment by garaemon** [comment]
+1

File: calibration_setup_helper/package.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<package>
+  <name>calibration_setup_helper</name>
+  <version>0.10.10</version>
+  <description>
+    This package contains a script to generate calibration launch and configurationfiles for your robot
+  </description>
+  <author>Michael Ferguson</author>
```
Comment (vrabaud) [inline calibration_setup_helper/package.xml]: please @k-okada  , set yourself as an author and maintainer.

**Comment by vrabaud** [comment]
also, please add a REAMD file that contains what you described at the beginning of that bug report.
@mikeferguson  , any opinion against ? Looks pretty useful to me

**Comment by mikeferguson** [comment]
I didn't look at it in detail, but it seems fine.

File: calibration_setup_helper/scripts/calibration_setup_helper.py [hunk 2]
```diff
@@ -0,0 +1,764 @@
+#!/usr/bin/env python
+
```
Comment (vrabaud) [inline calibration_setup_helper/scripts/calibration_setup_helper.py:2]: please add a license (may I recommend Apache 2.0 like anything OSRF does), you can copy paste from here:https://github.com/ros-aldebaran/romeo_robot/blob/master/romeo_description/scripts/generate_urdf.py

**Comment by vrabaud** [comment]
please set yourself as an author/maintainer. Installing your script would be more proper with a setup.py (especially to make your code runnable from Python3)

**Comment by k-okada** [comment]
ok, I have fexed, and also create #37 that include branch to use setup.py. Just to make sure that if se use setup.py , it install the script under global bin instead of package bin. Is it ok?

**Comment by vrabaud** [comment]
You are totally right, this is not ok :( I was wrong.
The setup.py helps dealing with Python 3 but let's ignore that for now. Just update the maintainers/authors and license on this pull request.
Sorry for the noise.

**Pull request merged**